examples/seismic/source.py

```diff
@@ -49,12 +57,47 @@ def __new__(cls, name, grid, ntime=None, npoint=None, data=None,
         # If provided, copy initial data into the allocated buffer
         if data is not None:
             obj.data[:] = data
+
+        # Set the origin and interval in the time axis
+        if time is None:
+            if dt is None or t0 is None:
+                error('Either time or t0 and dt are required to '
+                      'initialise source/receiver objects')
+            else:
+                obj.t0 = t0
+                obj.dt = dt
+        else:
+            obj.t0 = time[0]
+            obj.dt = time[1]-time[0]
+
         return obj
 
     def __init__(self, *args, **kwargs):
         if not self._cached():
             super(PointSource, self).__init__(*args, **kwargs)
 
+    def resample(self, dt, window=None):
```

Comment:
@property with setter instead of new obj?

Reply:
Same as previous comment.